onnxruntime/python/tools/transformers/models/stable_diffusion/benchmark.py

```diff
@@ -655,28 +857,225 @@ def warmup():
     image_filename_prefix = get_image_filename_prefix("trt", model_name, batch_size, disable_safety_checker)
 
     latency_list = []
-    prompts = example_prompts()
+    prompts, negative_prompt = example_prompts()
     for i, prompt in enumerate(prompts):
         if i >= num_prompts:
             break
         for j in range(batch_count):
             inference_start = time.time()
-            images = pipe(
+            # Use warmup mode here since non-warmup mode will save image to disk.
+            images, pipeline_time = pipeline.run(
                 [prompt] * batch_size,
-                num_inference_steps=steps,
-            ).images
+                [negative_prompt] * batch_size,
+                height,
+                width,
+                denoising_steps=steps,
+                guidance=7.5,
```

Comment:
Can the inference runs for measuring latency be generalized? After initializing the necessary resources and information, many of the runs appear to follow the format of `warmup --> first_run_memory --> second_run_memory --> measure_latencies_of_prompts(*args, **kwargs)` where `args` and `kwargs` are the needed parameters for the particular run. There appears to be initial support for this with the `warmup()` calls.

Reply:
yes, it is possible. However, that will change many lines of code so I would like to make the refactoring in another PR instead of mixing with this PR.